src/main/src/main/java/org/geoserver/catalog/impl/ResourceInfoImpl.java

```diff
@@ -241,30 +241,55 @@ public ReferencedEnvelope boundingBox() throws Exception {
             try {
-                nativeBox = getLatLonBoundingBox().transform(declaredCRS, true);
+                if (declaredCRS != null) {
+                    nativeBox = getLatLonBoundingBox().transform(declaredCRS, true);
+                } else {
+                    LOGGER.log(
+                            Level.WARNING,
+                            "Failed to derive native bbox, there is no declared CRS provided");
+                    return null;
+                }
             } catch (Exception e) {
                 LOGGER.log(Level.WARNING, "Failed to derive native bbox from declared one", e);
                 return null;
             }
+        } else if (nativeBox.getCoordinateReferenceSystem() == null) {
+            if (nativeCRS != null) {
+                // Ensure provided nativeBox using nativeCRS
+                nativeBox = ReferencedEnvelope.create(nativeBox, nativeCRS);
+            }
+        } else if (nativeCRS != null
+                && !CRS.equalsIgnoreMetadata(nativeBox.getCoordinateReferenceSystem(), nativeCRS)) {
+            LOGGER.log(Level.FINE, "The native bounding box srs does not match native crs");
+            nativeBox = ReferencedEnvelope.create(nativeBox, nativeCRS);
         }
 
-        ReferencedEnvelope result;
+        // make sure that in no case the actual field value is returned to the client, this
+        // is not a getter, it's a derivative, thus ModificationProxy won't do a copy on its own
         if (!CRS.equalsIgnoreMetadata(declaredCRS, nativeCRS)
                 && php == ProjectionPolicy.REPROJECT_TO_DECLARED) {
-            result = nativeBox.transform(declaredCRS, true);
+            if (nativeBox.getCoordinateReferenceSystem() == null) {
+                LOGGER.log(
+                        Level.WARNING,
+                        "Unable to reproject to declared crs (native bounding box srs and native crs are not defined)");
+                return null;
+            }
+            return nativeBox.transform(declaredCRS, true);
         } else if (php == ProjectionPolicy.FORCE_DECLARED) {
-            result = ReferencedEnvelope.create((Envelope) nativeBox, declaredCRS);
+            return ReferencedEnvelope.create((Envelope) nativeBox, declaredCRS);
         } else {
-            result = nativeBox;
+            if (nativeBox == null || nativeBox.getCoordinateReferenceSystem() == null) {
+                LOGGER.log(
+                        Level.WARNING,
+                        "Unable to determine native bounding crs (both native bounding box srs and native crs are not defined)");
+                // return null;
+            }
+            return ReferencedEnvelope.create(nativeBox);
         }
-
-        // make sure that in no case the actual field value is returned to the client, this
-        // is not a getter, it's a derivative, thus ModificationProxy won't do a copy on its own
-        return ReferencedEnvelope.create(result);
```

Comment:
This was an important cue to future developers, we should not lose it (I'm talking about the comment regarding ModificationProxy)

Reply:
The line is included further up